### ISSUE TITLE
[aws/aws-calico] Errors logged in calico-typha-horizontal-autoscaler

### ISSUE THREAD

**Issue by gazal-k** [aws/eks-charts#230]
I installed the latest version of chart aws/aws-calico;

```sh
NAME          	NAMESPACE  	REVISION	UPDATED                                 	STATUS  	CHART                	APP VERSION
aws-calico    	kube-system	9       	2020-08-16 12:08:12.123675565 +1000 AEST	deployed	aws-calico-0.3.0     	3.13.4     
```

And I see the following errors in the deployment `calico-typha-horizontal-autoscaler`

```sh
E0816 02:14:34.053996       1 reflector.go:125] github.com/kubernetes-incubator/cluster-proportional-autoscaler/pkg/autoscaler/k8sclient/k8sclient.go:96: Failed to list *v1.Node: Get https://172.20.0.1:443/api/v1/nodes: dial tcp 172.20.0.1:443: i/o timeout
E0816 02:14:37.052281       1 autoscaler_server.go:86] Error while getting cluster status: timed out waiting for the condition
E0816 02:14:47.052360       1 autoscaler_server.go:86] Error while getting cluster status: timed out waiting for the condition
E0816 02:14:57.052308       1 autoscaler_server.go:86] Error while getting cluster status: timed out waiting for the condition
```

I'm actually not very across what this deployment is supposed to do even.
An earlier version of this chart used to give errors like:

```sh
E0703 00:05:53.760651       1 autoscaler_server.go:120] Update failure: the server could not find the requested resource
E0703 00:06:03.953371       1 autoscaler_server.go:120] Update failure: the server could not find the requested resource
E0703 00:06:13.761335       1 autoscaler_server.go:120] Update failure: the server could not find the requested resource
E0703 00:06:23.760804       1 autoscaler_server.go:120] Update failure: the server could not find the requested resource
E0703 00:06:33.760613       1 autoscaler_server.go:120] Update failure: the server could not find the requested resource
E0703 00:06:43.951557       1 autoscaler_server.go:120] Update failure: the server could not find the requested resource
```

But calico's primary functionality of implementing `NetworkPolicy`s seemed to work just fine. But I would like to avoid these errors being logged all the time.
I had opened a support ticket about the errors and got feedback to install from https://github.com/aws/amazon-vpc-cni-k8s/blob/master/config/v1.6/calico.yaml.
But we have a GitOps based mechanism to manage our clusters and we rely on helmfile and helm for that. So using this helm chart would be ideal.

**Comment by gazal-k** [comment]
Looks like there was more to this latest version than just `calico-typha-horizontal-autoscaler` logging errors. `calico-node` pods logged the following:

```sh
2020-08-18 07:00:23.825 [INFO][9] startup.go 309: Early log level set to info
2020-08-18 07:00:23.825 [INFO][9] startup.go 325: Using NODENAME environment for node name
2020-08-18 07:00:23.825 [INFO][9] startup.go 337: Determined node name: ip-192-168-155-202.ap-southeast-2.compute.internal
2020-08-18 07:00:23.826 [INFO][9] startup.go 369: Checking datastore connection
2020-08-18 07:00:23.834 [INFO][9] startup.go 393: Datastore connection verified
2020-08-18 07:00:23.834 [INFO][9] startup.go 104: Datastore is ready
2020-08-18 07:00:24.023 [ERROR][9] startup.go 146: failed to query kubeadm's config map error=configmaps "kubeadm-config" is forbidden: User "system:serviceaccount:kube-system:calico-node" cannot get resource "configmaps" in API group "" in the namespace "kube-system"
2020-08-18 07:00:24.023 [WARNING][9] startup.go 1214: Terminating
Calico node failed to start
```

I had just prepared a fix for this by comparing with https://github.com/aws/amazon-vpc-cni-k8s/blob/master/config/v1.6/calico.yaml, but looks like somebody beat me to it: #228 :slightly_smiling_face:

**Comment by gazal-k** [comment]
this issue has been resolved in `aws-calico-0.3.1`
